File: homework_1/11.draw_home/draw_home.java
import java.util.Scanner;

class draw_home {
    public static void main(String[] args){
        Scanner input = new Scanner(System.in);

        int width = input.nextInt();
        int height = width / 5;

        draw_roof(width);
        draw_line(width, '_', true);
        draw_side_wall(height,width);
        draw_line(width, '_', true);

   
        input.close();
    }

    public static void draw_line(int width,char symbol, boolean newline){
        for(int i = 0; i < width; ++i){
            System.out.print(symbol);
        }
        if (newline) System.out.println();
    }

    public static void draw_side_wall(int height, int width){
        for (int i = 0; i < height; ++i){
            System.out.println();
            System.out.print('|');
            draw_line(width - 2, ' ', false);
            System.out.println('|');
        }
    }

    public static void draw_roof(int width){
        for(int i = 1; i <= width; i+=2){
            draw_line((width - i) / 2, ' ', false);
            draw_line((i), '*', true);
        }
    }
}

File: homework_1/11.draw_home/draw_home.py
def draw_line(width, symbol, newline):
    print(width * symbol, end="")
    if newline: print()

def draw_side_wall(height, width):
    for level in range(height):
        print("\n|", end="")
        draw_line(width - 2, " ", False)
        print("|")

def draw_roof(width):
    for i in range(1, width + 1, 2):
        draw_line((width - i) // 2, " ", False)
        draw_line(i, "*", True)

width = int(input())
height = width // 5 + 1

draw_roof(width)
draw_line(width, '_', True)
draw_side_wall(height, width)
draw_line(width, '_', True)